lib/msal-common/src/client/SilentFlowClient.ts

```diff
@@ -39,8 +41,34 @@ export class SilentFlowClient extends BaseClient {
         request: CommonSilentFlowRequest
     ): Promise<AuthenticationResult> {
         try {
-            return await this.acquireCachedToken(request);
+            const cachedToken = await this.acquireCachedToken(request);
```

Comment:
There is concept of [typescript tuples](https://www.w3schools.com/typescript/typescript_tuples.php). I can see how this would make sense, however, all the unit tests using acquireCachedToken would need to updated to account for the tuple return type. I think that would be fine. Thoughts?

Reply:
Yes, do it https://levelup.gitconnected.com/the-boy-scouting-rule-a-simple-yet-effective-programming-principle-5d504c7dd993#:~:text=The%20Boy%20Scouting%20Rule%20is,environment%20cleaner%20than%20they%20found.

Reply:
The issue with using lastCacheOutcome like currently is if one forgets to set/reset it, it can create bugs, which can be difficult to track down. Since Bogdan gave you the blessing on refactoring, I would just put it in a separate PR, you can use this branch as the base and then merge right after this one. (Smaller and more focused PRs are easier to review and track.)

Hmm, I knew about the rule; didn't know what it was called...

Reply:
Just make sure the `acquireToken` call is handled properly as I think we use that API in `msal-browser`. Also may be `cachedToken` is not named correctly, the return type is `AuthenticationResult` and not just a `token`.